lib/raygun.dart

```diff
@@ -0,0 +1,226 @@
+    // Convert given values to native friendly types
+    Pointer<Int8> _convoId = conversationId.toNativeUtf8().cast<Int8>();
+    Pointer<Int8> _messageId = messageId.toNativeUtf8().cast<Int8>();
+    G_FFIVec_String _messages = messages.value.cast().elementAt(0);
+    int lines = _messages.len;
+    // Invoke and result check
+    G_FFIResult_Null result = bindings.raygun_send(
+        pRaygun, _convoId, _messageId, _messages.ptr.elementAt(0), lines);
+    if (result.error.address.toString() != "0") {
+      throw WarpException(result.error);
+    }
+  }
+
+  delete(String conversationId, [String? messageId]) {
+    // Convert given values to native friendly types
+    Pointer<Int8> _convoId = conversationId.toNativeUtf8().cast<Int8>();
+    Pointer<Int8> _messageId =
+        messageId != null ? messageId.toNativeUtf8().cast<Int8>() : nullptr;
+    // Invoke and result check
+    G_FFIResult_Null result =
+        bindings.raygun_delete(pRaygun, _convoId, _messageId);
+    if (result.error.address.toString() != "0") {
+      throw WarpException(result.error);
+    }
+  }
+
+  react(String conversationId, String messageId, ReactionState reactionState,
+      String emoji) {
+    // Convert given values to native friendly types
+    Pointer<Int8> _convoId = conversationId.toNativeUtf8().cast<Int8>();
+    Pointer<Int8> _messageId = messageId.toNativeUtf8().cast<Int8>();
+    int _reactionState = reactionState.index;
+    Pointer<Int8> _emoji = emoji.toNativeUtf8().cast<Int8>();
+    // Invoke and result check
+    G_FFIResult_Null result = bindings.raygun_react(
+        pRaygun, _convoId, _messageId, _reactionState, _emoji);
+    if (result.error.address.toString() != "0") {
+      throw WarpException(result.error);
+    }
+  }
+
+  pin(String conversationId, String messageId, PinState pinState) {
+    // Convert given values to native friendly types
+    Pointer<Int8> _convoId = conversationId.toNativeUtf8().cast<Int8>();
+    Pointer<Int8> _messageId = messageId.toNativeUtf8().cast<Int8>();
+    int _pinState = pinState.index;
+    // Invoke and result check
+    G_FFIResult_Null result =
+        bindings.raygun_pin(pRaygun, _convoId, _messageId, _pinState);
+    if (result.error.address.toString() != "0") {
+      throw WarpException(result.error);
+    }
+  }
+
+  reply(String conversationId, String messageId, Message messages) {
+    // Convert given values to native friendly types
+    Pointer<Int8> _convoId = conversationId.toNativeUtf8().cast<Int8>();
+    Pointer<Int8> _messageId = messageId.toNativeUtf8().cast<Int8>();
+    G_FFIVec_String _messages = messages.value.cast().elementAt(0);
+    int lines = _messages.len;
+    // Invoke and result check
+    G_FFIResult_Null result = bindings.raygun_reply(
+        pRaygun, _convoId, _messageId, _messages.ptr.elementAt(0), lines);
+    if (result.error.address.toString() != "0") {
+      throw WarpException(result.error);
+    }
+  }
+
+  ping(String conversationId) {
```

Comment:
This function can be removed. Overlooked this but it is no longer needed.

Reply:
Removed.